### PR TITLE
Keep updating CHANGES.md.

### PR DESCRIPTION
**Pull request by ltratt**
@ratmice do you think there will be further semver-breaking changes? If so, I'm happy to keep waiting to make a release (but I'll keep updating CHANGES.md intermittently while I vaguely remember what's happened.)

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -19,6 +46,14 @@
     pub fn add_rule(&mut self, (name, name_span): (String, Span), actiont: Option<String>) {
   ```
 
+* `YaccParserError` has been renamed to `YaccGrammarError` and changed from an
+  `enum` (which frequently changed, breaking semver) to a `struct`. The `kind`
+  field is now internal (so won't break semver as often).
```
Comment (ratmice) [inline CHANGES.md:51]: There is also some removals associated with this which should probably be noted: 
Perhaps something to the effect of:

 `GrammarValidationError` and `YaccParserError` have been combined as `YaccGrammarError`, replacing the enum `YaccGrammarError` which had a variant for each of those.

File: CHANGES.md [hunk 2]
```diff
@@ -19,6 +46,14 @@
...
+
+* `LexBuildresult` returns on failure `Err(Vec<LexBuildError>)` rather than
+  `Err(LexBuildError)`.
+
```
Comment (ratmice) [inline CHANGES.md:55]: This one I still need to look at and see if we need to give `LexBuildResult` the same treatment regarding `spans` field and `kind` as we did to `YaccGrammarError`, so there may be one more semver change here perhaps.

Comment (ltratt) [reply CHANGES.md:55]: Ah, OK. Should I delete this line or ... ?

Comment (ratmice) [reply CHANGES.md:55]: No I think what is written shouldn't be changing, 
Just wanted to point out one last semver change might be needed.

Comment (ltratt) [reply CHANGES.md:55]: Got it!

File: CHANGES.md (hunk 1 above)
Comment (ltratt) [reply CHANGES.md]: Fixed in 0b1d465.

**Comment by ratmice** [comment]
Please squash

**Comment by ltratt** [comment]
Squashed.

**Pull request merged**